koku/masu/processor/ocp/ocp_cost_model_cost_updater.py

```diff
@@ -175,6 +170,130 @@ def _update_markup_cost(self, start_date, end_date):
             accessor.populate_markup_cost(markup, start_date, end_date, self._cluster_id)
         LOG.info("Finished updating markup.")
 
+    def _build_node_tag_rate_case_statement_str(self, rate_dict, start_date, default_rate_dict={}, unallocated=False):
+        """Given a tag key, value, and rate return a CASE SQL statement for tag based monthly SQL."""
+        case_dict = {}
+        cpu_distribution_term = """
+            sum(pod_effective_usage_cpu_core_hours)
+                / max(node_capacity_cpu_core_hours)
+        """
+        memory_distribution_term = """
+            sum(pod_effective_usage_memory_gigabyte_hours)
+                / max(node_capacity_memory_gigabyte_hours)
+        """
+
+        if unallocated is True:
+            cpu_distribution_term = """
+                (max(node_capacity_cpu_core_hours) - sum(pod_effective_usage_cpu_core_hours))
+                    / max(node_capacity_cpu_core_hours)
+            """
+            memory_distribution_term = """
+                (max(node_capacity_memory_gigabyte_hours) - sum(pod_effective_usage_memory_gigabyte_hours))
+                    / max(node_capacity_memory_gigabyte_hours)
+            """
+
+        for tag_key, tag_value_rates in rate_dict.items():
+            cpu_statement_list = ["CASE"]
+            memory_statement_list = ["CASE"]
+            for tag_value, rate_value in tag_value_rates.items():
+                label_condition = f"pod_labels->>'{tag_key}'='{tag_value}'"
+                if unallocated is True:
+                    label_condition_list = ["pod_labels = '{", f'"{tag_key}": "{tag_value}"', "}'"]
+                    label_condition = "".join(label_condition_list)
```

Comment:
This can be a single format string which returns a sting like this: `pod_labels = '{"tk": "tv"}'`
```suggestion
                    label_condition_list = ["pod_labels = '{", f'"{tag_key}": "{tag_value}"', "}'"]
                    label_condition = f'pod_labels = \'{{"{tag_key}": "{tag_value}"}}\''
```